projects/onto-search/src/lib/onto-search-facet/date-range-facet/date-range-facet.component.ts

```diff
@@ -178,69 +196,27 @@ export class DateRangeFacetComponent extends OntoSearchFacetComponent implements
         .reduce((accumulator, count) => accumulator + count, 0);
   }
 
-  private addTooltipsToCalendarCells(view: string, datesMap: Map<string, number>, dateFormat: string): void {
-    const calendarCells = this.getCalendarCellsForView(view);
-    calendarCells.forEach((calendarCell) => {
-      let found = false;
-      datesMap.forEach((count, dateString) => {
-        const calendarCellDateString = moment(calendarCell.getAttribute('aria-label'), dateFormat);
-        if (calendarCellDateString.isSame(dateString)) {
-          found = true;
-          this.createTooltipForCalendarCell(calendarCell, count);
-        }
+  private subscribeToBtnListeners(buttons: any): void {
+    buttons.forEach((btn) => {
+      const unsubscribe = this.renderer.listen(btn, 'click', () => {
+        this.calendarOpened();
       });
-      if (!found) {
-        this.removeTooltipFromCalendarCell(calendarCell);
-      }
+      this.buttonUnsubscribeFns.push(unsubscribe);
```

Comment:
I kind of prefer to have this operation to be atomic. You can iterate buttons, subscribe the listeners, collect unsubscribe callbacs in array and push them at once with the array spread operator in the buttonUnsubscribeFns array. This way it won't be possible to leak them if in the meanwhile unsubscribe is called and not everything is pushed there. I guess this is not realistic to happen, but still, its better to prevent such scenarios.